commands/publish_dafsa.py

```diff
@@ -0,0 +1,123 @@
+import requests
+from requests.exceptions import HTTPError
+import tempfile
+import subprocess
+from kinto_http import Client, KintoException
+import os
+
+COMMIT_HASH_URL = "https://api.github.com/repos/publicsuffix/list/commits/master"
+
+LIST_URL = (
+    "https://raw.githubusercontent.com/publicsuffix/list/master/public_suffix_list.dat"
+)
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"
+
+
+SERVER = "https://kinto.dev.mozaws.net/v1"
+CREDENTIALS = (os.getenv("USERNAME"), os.getenv("PASSWORD"))  # (username, password)
+
+BUCKET_ID = "firefox-core-network-dns"
+COLLECTION_ID = "public-suffix-list"
+RECORD_ID = "latest-commit-hash"
+
+
+def handle_request_errors(func):
+    def inner(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Fetching Failed")
+
+    return inner
+
+
+@handle_request_errors
+def get_latest_hash():
+    response = requests.get(COMMIT_HASH_URL)
+    response.raise_for_status()
+
+    modified_files = response.json()["files"]
+    for f in modified_files:
+        if f["filename"] == "public_suffix_list.dat":
+            return response.json()["sha"]
```

Comment:
I don't think this will work. The URL returns only one commit, the latest one (master).
The URL should return the list of commits for this particular file. `/commits?path=...`